app/models/miq_queue.rb

```diff
@@ -541,6 +543,13 @@ def self.get_worker(task_id)
 
   private
 
+  def activate_miq_task(args)
+    MiqTask.update_status(miq_task_id, MiqTask::STATE_ACTIVE,  MiqTask::STATUS_OK, "Task starting") if miq_task
```

Comment:
Prefer `if miq_task_id`, so it doesn't need to go out and fetch the task on every dispatch.  That being said, if the task is deleted out from underneath the queue item, the MiqTask.update_status method should probably handle that.

Reply:
👍